### ISSUE TITLE
Panic when block is empty

### ISSUE THREAD

**Issue by Ianwu0812** [terraform-providers/terraform-provider-azurerm#3688]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.12.2
+ provider.azurerm v1.30.1

### Affected Resource(s)

* azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp

provider "azurerm" {
}

resource "azurerm_resource_group" "rg1" {
    name     = "${var.names.rg}"
    location = "${var.location}"

    tags = {
        environment = "development"
    }
}
resource "azurerm_kubernetes_cluster" "cluster1" {
    name = "${var.names.aks}"
    location = "${var.location}"
    resource_group_name = "${var.names.rg}"
    dns_prefix          = "aks-dev"
    agent_pool_profile {
        name            = "default"
        count           = 2
        vm_size         = "Standard_D4s_v3"
        os_type         = "Linux"
        os_disk_size_gb = 30
    }
    service_principal {
        client_id     = "${var.client_id}"
        client_secret = "${var.client_secret}"
    addon_profile {
        #oms_agent {}
    }
    depends_on = [azurerm_resource_group.rg1]
}

```

### Panic Output

panic: interface conversion: interface {} is nil, not map[string]interface {}

### Expected Behavior

Show detailed message about invalid block, then user will know about what should do.

### Actual Behavior

Just show the crash log

### Steps to Reproduce

1. `terraform apply`

Yes, just apply the config I provided.

### References

It will occur panic error, fixed after delete block

    addon_profile {
        #oms_agent {}
    }

* #0000

**Comment by NickMetz** [comment]
Hi @Ianwu0812,
i was running in same issue and worked around like this:

```
variables.tf

variable "http_application_routing_enabled" {
  description = "Enable or disable http_application_routing"
  default     = false
}

variable "oms_agent" {
  type        = list(map(string))
  description = "A oms_agent block support."
  default     = []
}

main.tf
[..]
  addon_profile {
    http_application_routing {
      enabled = var.http_application_routing_enabled
    }

    dynamic "oms_agent" {
      for_each = var.oms_agent
      content {
        enabled                    = lookup(oms_agent.value, "enabled", null)
        log_analytics_workspace_id = lookup(oms_agent.value, "log_analytics_workspace_id", null)
      }
    }
  }

```
So i use http_application_routing as placeholder in addon_profile which is default set to false. This will prevent crashing in case addon_profile is empty. I know this is not a great solution but just in case it is blocking you.

**Comment by Ianwu0812** [comment]
Hi @NickMetz, 
Thanks for your advice, it's useful for me!

**Comment by tombuildsstuff** [comment]
Fixed via #4898
